### ISSUE TITLE
mkdir on CEPH radosgw doesn't work properly

### ISSUE THREAD

**Issue by dirkjanw** [s3fs-fuse/s3fs-fuse#363]
Hi,

when you try to create a directory on a mounted bucket from CEPH radosgw 0.94.5 and S3FS (either latest GIT or 1.79 release), it will give an input/output error and a subsequent ls returns a file:

```
root@frontend-dev:/mnt/ceph-01/testbucket# ls -la
total 5
drwx------ 1 root root    0 Jan  1  1970 .
drwxr-xr-x 5 root root 4096 Feb  7 21:07 ..
-rwxr-xr-x 1 root root    0 Feb 12 14:55 test
root@frontend-dev:/mnt/ceph-01/testbucket# mkdir test2
mkdir: cannot create directory âtest2â: Input/output error
root@frontend-dev:/mnt/ceph-01/testbucket# ls -la
total 6
drwx------ 1 root root    0 Jan  1  1970 .
drwxr-xr-x 5 root root 4096 Feb  7 21:07 ..
-rwxr-xr-x 1 root root    0 Feb 12 14:55 test
-rwxr-xr-x 1 root root    0 Feb 12 15:01 test2
root@frontend-dev:/mnt/ceph-01/testbucket#
```

Please find the debugging log attached!

[s3fs.txt](https://github.com/s3fs-fuse/s3fs-fuse/files/128086/s3fs.txt

**Comment by dirkjanw** [comment]
While looking through the closed issues, I found #358 which looks quite the same. Upon inspecting the headers and the source-code, it looks like the radosgw embedded webserver is returning 'Content-type' instead of 'Content-Type' (capital T).

If I interpret the RFC https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2 right, these headers should not be interpreted case sensitive, so while not 'standard' this reply isn't wrong either, but correct me if I'm wrong :)

I will see if the radosgw guys are inclined to change the response, however perhaps you'd like to change the way of checking headers too :)

**Comment by ggtakec** [comment]
@dirkjanw I merged codes by #364, it fixed this issue.
Please use latest codes in master branch.
Regards,
